modules/client/src/types.ts

```diff
@@ -46,15 +47,14 @@ export interface ClientOptions {
 export type InternalClientOptions = ClientOptions & {
   appRegistry: AppRegistry;
   cfCore: CFCore;
+  config: GetConfigResponse;
   contract?: MultisigState;
+  ethProvider: providers.JsonRpcProvider;
   messaging: IMessagingService;
   multisigAddress: string;
   network: utils.Network; // TODO: delete! use bos branch!
   node: NodeApiClient;
-  nodePublicIdentifier: string;
-  ethProvider: providers.JsonRpcProvider;
   store: CFCoreTypes.IStoreService;
-  contractAddresses: ContractAddresses;
```

Comment:
nodePublicIdentifier and contractAddresses are available via config, if we pass the whole config object in then we don't need these separately.